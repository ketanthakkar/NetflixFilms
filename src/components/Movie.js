import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router';
import PropTypes from 'prop-types';
import Title from './Title';
import Footer from './Footer';
import Content from './Content';
import { fetchMovieDetail } from '../actions';

class Movie extends Component {
  static fetchData(dispatch, match) {
    return dispatch(fetchMovieDetail(match.params.id));
  }

  componentDidMount() {
    if (this.props.match) {
      this.props.fetchMovieDetail(this.props.match.params.id);
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.match) {
      const { id } = this.props.match.params;
      if (id !== prevProps.match.params.id) {
        this.props.fetchMovieDetail(this.props.match.params.id);
      }
    }
  }

  render() {
    const { movie } = this.props;
    return (
            <React.Fragment>
                { movie
                && <article className="movie-container">
                    <header className="header-section">
                        <Title />
                    </header>
                    <section className="movie-poster">
                        <img className="movie-img" src={movie.poster_path} alt={movie.tagline} />
                        <div className="movie-detail">
                            <span className="movie-title">{movie.title.toUpperCase()}</span>
                            <span className="movie-genres">{movie.genres.join(' & ')}</span>
                            <div>
                                <span className="movie-year">{movie.release_date}</span>
                                <span className="movie-length">{`${movie.runtime} min`}</span>
                            </div>
                            <p className="movie-description">{movie.overview}</p>
                        </div>
                    </section>
                    <Content showSearch={false} />
                    <Footer />
                </article> }
            </React.Fragment>
    );
  }
}

Movie.propTypes = {
  fetchMovieDetail: PropTypes.func,
  movie: PropTypes.object,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }),
}

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchMovieDetail,
}, dispatch);

const mapStateToProps = state => ({
  movie: state.movies.movie,
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Movie));