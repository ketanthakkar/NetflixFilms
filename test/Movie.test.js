import React from 'react';
import { shallow } from 'enzyme';
import Movie, { getMovie, mapStateToProps } from '../src/components/Movie';

describe('should render Movie correctly', () => {
  const movieData = {
    id: 337167,
    title: 'Fifty Shades Freed',
    tagline: "Don't miss the climax",
    vote_average: 6.1,
    vote_count: 1195,
    release_date: '2018-02-07',
    poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
    overview: 'Believing they have left behind shadowy figures from their past',
    budget: 55000000,
    revenue: 136906000,
    genres: [
      'Drama',
      'Romance',
    ],
    runtime: 106,
  };
  const tree = shallow(<Movie movie={ movieData }/>);
  test('Snapshot test', () => {
    expect(tree).toMatchSnapshot();
  });

  test('should get movie', () => {
    const state = {
      movies: "star",
    };
    const expected = "star";

    expect(getMovie(state)).toEqual(expected);
  })

  test('should map state to props', () => {
    const selector = mapStateToProps.resultFunc;
    const expected = {
      movie: 'star',
    };

    expect(selector({ movie: 'star' })).toEqual(expected);
  })

});
