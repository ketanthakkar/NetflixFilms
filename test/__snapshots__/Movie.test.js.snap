// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Movie correctly Snapshot test 1`] = `
<Fragment>
  <article
    className="movie-container"
  >
    <header
      className="header-section"
    >
      <Title />
    </header>
    <section
      className="movie-poster"
    >
      <img
        alt="Don't miss the climax"
        className="movie-img"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
      <div
        className="movie-detail"
      >
        <span
          className="movie-title"
        >
          FIFTY SHADES FREED
        </span>
        <span
          className="movie-genres"
        >
          Drama & Romance
        </span>
        <div>
          <span
            className="movie-year"
          >
            2018-02-07
          </span>
          <span
            className="movie-length"
          >
            106 min
          </span>
        </div>
        <p
          className="movie-description"
        >
          Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
        </p>
      </div>
    </section>
    <Connect(Content)
      showSearch={false}
    />
    <Footer />
  </article>
</Fragment>
`;
